agent/api/task/task.go

```diff
@@ -846,6 +876,81 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 	return hostConfig, nil
 }
 
+func (task *Task) awslogsEndpointOverride(hostConfig *docker.HostConfig, dockerVersion string) bool {
+	if hostConfig.LogConfig.Type != awslogsType {
+		return false
+	}
+
+	isVersionMatch := false
+	if task.isDockerCEVersion(dockerVersion) {
```

Comment:
You don't need to have this much logic; you can do it like this:

```go
match, err := utils.Version(">=19.03.1,>=19.03.1-ce,>=19.03.1-ee")
```

Or, alternately (and I think better!) would be to fix `utils.Version` (or maybe `utils.DockerVersion`) so that it can treat `-ce` and `-ee` as satisfying a bare number comparison (so `19.03.1-ce` would satisfy `>=19.03.1`).

Reply:
Correct me if I am wrong, if the minimum docker ee and docker ce share same versions we can definitely do this, but when they have different versions the logic cannot be simplified this way.

Let's say our minimum docker ce version is `19.03.1-ce`, and minimum docker ee version is `20.0.0-ee`. When our current docker version is `19.05.1-ee`, case `">=19.03.1,>=19.03.1-ce,>=20.0.0-ee"` will be matched, because `Matches` method will return true as `19.05.1-ee` is greater than `19.03.1-ce`. But actually this case should not be matched.

I am unsure about those versions right now, so I'll leave it as it is until we know them.

Reply:
I'd rather have our `utils.Version` (or `utils.DockerVersion`) fixed to treat both `-ce` and `-ee` as satisfying a bare number comparison.  `utils.Version` was written to compare SemVer version numbers, but Docker releases don't follow SemVer; `-ce` and `-ee` are not prereleases even though our matcher currently treats them as such.